.github/workflows/ci.yaml

```diff
@@ -68,8 +68,26 @@ jobs:
       - uses: Swatinem/rust-cache@v1
       # Needed for agent
       - run: cargo fmt --all -- --check
+      - run: rustup target add --toolchain nightly aarch64-apple-darwin
       - run: cargo clippy -- -D warnings
 
+  lint_macos:
+    runs-on: macos-latest
+    needs: changed_files
+    if: ${{needs.changed_files.outputs.rs_changed == 'true'}}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly
+          components: rustfmt, clippy
+          target: aarch64-apple-darwin
+      - uses: Swatinem/rust-cache@v1
+      # only mirrord, mirrord-sip contain macOS specific, so run clippy on that.
```

Comment:
Would it be notably slower to run it on all the code? That way we wouldn't have to remember to add stuff to here if we add MacOS-specific code in the future.

Reply:
Problem is not all code compiles for macOS (agent)..

Reply:
(It can't compile because of dependencies)